letsencrypt/rootfs/etc/services.d/lets-encrypt/run

```diff
@@ -276,7 +276,7 @@ elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-transip" ]; then
     bashio::config.require 'dns.transip_username'
     bashio::config.require 'dns.transip_api_key'
     if bashio::config.has_value 'dns.transip_global_key'; then
-    "dns_transip_global_key=$(bashio::config 'dns.transip_global_key')"
+    echo "dns_transip_global_key = $(bashio::config 'dns.transip_global_key')" >> "/data/dnsapikey"
```

Comment:
I think indentation is not correct here.

In any case, those optional settings we usually write in `file-structure.sh`. I think this should be the case with this value too.

Reply:
Indentation is wrong,  thanks for the review.

If the global_key is present, it is written in file-structure.sh, but as the comments of the issue 3876, this is not working.
I was also following other cases like gandi, which is writing also in the run script.

Please advise.

Reply:
The problem is, the config `transip_global_key` got introduced later in #3855, and is an optional config As in, transip can be used also without `transip_global_key`. However, users which had that provider setup do not have that config set.

But `file-structure.sh` assumes that the config is always present.

So what needs to happen here is that we check if `transip_global_key` exists, and only if it does it should be written out to the config file.

PR #3862 was on the right track, but somehow the author disappeared.

Btw, thanks a lot for picking this up!